spec/unit/configurer_spec.rb

```diff
@@ -905,22 +906,19 @@ def expects_neither_new_or_cached_catalog
   describe "when attempting failover" do
     it "should not failover if server_list is not set" do
       Puppet.settings[:server_list] = []
-      expect(configurer).not_to receive(:find_functional_server)
       configurer.run
     end
 
     it "should not failover during an apply run" do
       Puppet.settings[:server_list] = ["myserver:123"]
-      expect(configurer).not_to receive(:find_functional_server)
       catalog = Puppet::Resource::Catalog.new("tester", Puppet::Node::Environment.remote(Puppet[:environment].to_sym))
-      configurer.run :catalog => catalog
+      configurer.run(catalog: catalog)
```

Comment:
Yeah... previously it was asserting that a private method was never called, which is very fragile. We might rename the `Configurer#find_functional_server` method and regress on the functionality, yet this test would still pass. So instead, I removed the "not_to receive" line and I'm relying on webmock to raise if the code ever makes a network connection (since we didn't register an allowable one via `stub_request`).

Instead I'm relying on webmock to raise if we ever were to make a network connection to `myserver`.

Reply:
Huh, that still doesn't totally make sense to me,  but hopefully one day we can find the sense :)